### PR TITLE
Add StringSet metrics to Python SDK

### PR DESCRIPTION
**Pull request by Abacn**
**Please** add a meaningful description for your change here

Direct runner and Flink runner (loopback) working. For Dataflow runner, needs to regenerate apiclient, will be done in a separate PR

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
R: @robertwb

File: sdks/python/apache_beam/metrics/cells.py [hunk 1]
```diff
@@ -553,3 +610,22 @@ def combine(self, x, y):
...
+    return set()
+
+  def combine(self, x, y):
+    # type: (set, set) -> set
+    if len(x) == 0:
+      return y
+    elif len(y) == 0:
+      return x
+    else:
+      return set.union(x, y)
```
[the remark this answers is not in the document]
Comment (Abacn) [reply sdks/python/apache_beam/metrics/cells.py:628]: this mirrors #31803 . I wonder in most cases the aggregation encounters one empty and one non-empty cells, so a short cut optimization is reasonable here.

> Could there be any concerns about side effects due to Aggregation returning a mutable collection

This is a good point. I was hesitating if I should return set / frozenset . However in Python frozenset isn't instance of set (`isinstance(frozenset(), set)` is False) while Java ImmutableSet is Set, I ended up with set. Do you think we should make the result of StringSet query a frozenset instance everywhere?

File: sdks/python/apache_beam/metrics/cells.py [hunk 2]
```diff
@@ -268,6 +268,63 @@ def to_runner_api_monitoring_info_impl(self, name, transform_id):
...
+  def combine(self, other):
+    # type: (StringSetCell) -> StringSetCell
+    combined = StringSetAggregator().combine(self.data, other.data)
+    result = StringSetCell()
+    result.data = combined
+    return result
+
+  def to_runner_api_monitoring_info_impl(self, name, transform_id):
+    from apache_beam.metrics import monitoring_infos
+    # return monitoring_infos.set
```
[the remark this answers is not in the document]
Comment (Abacn) [reply sdks/python/apache_beam/metrics/cells.py]: thanks for catching it, fixed

File: sdks/python/apache_beam/metrics/monitoring_infos.py [hunk 3]
```diff
@@ -67,10 +72,12 @@
     common_urns.monitoring_info_types.DISTRIBUTION_INT64_TYPE.urn)
 LATEST_INT64_TYPE = common_urns.monitoring_info_types.LATEST_INT64_TYPE.urn
 PROGRESS_TYPE = common_urns.monitoring_info_types.PROGRESS_TYPE.urn
+SET_STRING_TYPE = common_urns.monitoring_info_types.SET_STRING_TYPE.urn
```
[the remark this answers is not in the document]
Comment (Abacn) [reply sdks/python/apache_beam/metrics/monitoring_infos.py]: fixed. There is already an inconsistency between proto "SET_STRING_TYPE" and "STRING_SET_TYPES". Try to keep names inside Python SDK now

File: sdks/python/apache_beam/runners/portability/portable_runner.py [hunk 4]
```diff
@@ -437,15 +437,16 @@ def _combine(committed, attempted, filter):
     ]
 
   def query(self, filter=None):
-    counters, distributions, gauges = [
+    counters, distributions, gauges, stringsets = [
```
[the remark this answers is not in the document]
Comment (Abacn) [reply sdks/python/apache_beam/runners/portability/portable_runner.py:440]: Now use string_set / string_sets

File: sdks/python/apache_beam/metrics/cells.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tvalentyn) [reply sdks/python/apache_beam/metrics/cells.py:627]: I think it depends on the lifecycle of these sets; given Java implementation, my guess is that it is probably not a concern.

**Review by tvalentyn — APPROVED** [review]

**Comment by Abacn** [comment]
PreCommit YAML Xlang Direct fails with slf4jlogger classDefNotFoundError. Test passed on an earlier commit and the change later on are trivial naming changes. Likely bad gradle cache and merging for now

**Pull request merged**